### ISSUE TITLE
Presets tie the output head where the reference model doesn't; OLMo_7B also uses OLMo's fused MLP width

### ISSUE THREAD

**Issue by mira687** [openlanguagemodel/openlanguagemodel#83]
Building the named presets on the `meta` device makes their parameter counts checkable in seconds, with no weights, no download and no GPU:

```python
import torch
from olm.models import Llama3_1_405B
with torch.device("meta"):
    m = Llama3_1_405B()
print(f"{sum(p.numel() for p in m.parameters()):,}")   # 403,752,042,496
```

Six families come out low by exactly one embedding matrix each, and `OLMo_7B` comes out 4.1B high for an unrelated reason.

### 1. `tie_weights` defaults to `True`, and the older presets never override it

`Llama2Model`, `Llama3Model`, `Qwen2Model`, `Phi3Model`, `Phi4Model` and `OLMoModel` all declare `tie_weights: bool = True`, and none of their named presets pass anything else. The newer families do the opposite: `Qwen3`, `Qwen3Next`, `Qwen3_5`, `Olmo3`, `DeepSeekV3`, `Llama4Maverick`, `MiniMaxM2`, `MiniMaxM2_5`, `MistralLarge3`, `MistralSmall3_1`, `KimiK2`, `KimiLinear`, `Nanbeige4_1`, `Sarvam` and `Step3_5` each pass `tie_weights=False` explicitly, several with a docstring line saying the named preset does so. So the intent reads as fidelity to the reference config, and these are the presets that predate it.

`tie_word_embeddings` as published, read from the configs rather than recalled:

| reference config | tie | OLM preset | |
|---|---|---|---|
| `NousResearch/Llama-2-7b-hf`, `Llama-2-70b-hf` | `false` | `Llama2_7B/13B/70B` (13B not read) | ties — mismatch |
| `Meta-Llama-3.1-8B`, `Meta-Llama-3.1-70B` | `false` | `Llama3_1_8B/70B/405B` (405B not read) | ties — mismatch |
| `Qwen/Qwen2.5-7B`, `Qwen2.5-72B` | `false` | `Qwen2_5_7B/14B/32B/72B` (14B, 32B not read) | ties — mismatch |
| `microsoft/Phi-3.5-mini-instruct` | `false` | `Phi3_5_Mini` | ties — mismatch |
| `microsoft/phi-4` | `false` | `Phi4_14B` | ties — mismatch |
| `allenai/OLMo-7B` (`weight_tying`) | `false` | `OLMo_7B` | ties — mismatch |
| `unsloth/Llama-3.2-1B` | `true` | `Llama3_2_1B/3B` | ties — correct |
| `Qwen/Qwen2.5-0.5B` | `true` | `Qwen2_5_0_5B/1_5B/3B` | ties — correct |
| `unsloth/gemma-3-27b-pt` (key absent → HF default `true`) | `true` | `Gemma2*`, `Gemma3*` | ties — correct |
| `facebook/opt-125m` (key absent → HF default `true`) | `true` | `OPT125M` | ties — correct |
| GPT-2 | `true` | `GPT2*` | ties — correct |

### 2. Untying puts them back on the reference numbers

| preset | ships as | + vocab x embed | untied |
|---|---|---|---|
| `Llama3_1_8B` | 7,504,924,672 | 128256 x 4096 | **8,030,261,248** (8.03B) |
| `Llama3_1_70B` | 69,503,033,344 | 128256 x 8192 | **70,553,706,496** (70.55B) |
| `Llama3_1_405B` | 403,752,042,496 | 128256 x 16384 | **405,853,388,800** (405.85B) |
| `Llama2_7B` | 6,607,343,616 | 32000 x 4096 | **6,738,415,616** (6.74B) |
| `Qwen2_5_7B` | 7,070,619,136 | 152064 x 3584 | **7,615,616,512** |
| `Phi4_14B` | 14,145,704,960 | 100352 x 5120 | **14,659,507,200** |

The three Llama 3.1 numbers are the familiar 8.03B / 70.55B / 405.85B to the digit, which is the part that makes this a fidelity bug rather than a taste question.

### 3. `OLMo_7B` has a second problem: `intermediate_size=22016`

22016 is OLMo's *fused* MLP width. `allenai/OLMo-7B`'s own config has `mlp_hidden_size: 22016`, and the HF port `allenai/OLMo-7B-hf` records the per-branch equivalent as `intermediate_size: 11008`. `SwiGLUFFN` already allocates `up_proj: embed_dim -> 2 * intermediate_size`, so passing the fused number doubles the FFN:

| `intermediate_size` | `tie_weights` | params |
|---|---|---|
| 22016 | `True` | 11,010,473,984 — what ships |
| 11008 | `True` | 6,681,952,256 |
| 11008 | `False` | **6,887,899,136** — i.e. 6.89B |

Both fixes apply to that one preset, and `allenai/OLMo-7B` sets `weight_tying: false` directly, so no inference from the HF port is needed for the tie half.

### Not checked

- `Phi3_Small`: its config omits `tie_word_embeddings` and ships remote code, so I did not want to infer the default.
- Whether anything under `tests/` or the parity work depends on the current counts.
- OLMo's `vocab_size: 50280` vs `embedding_size: 50304` padding — the preset uses 50280; using 50304 adds 196,608 params on top of the 6,887,899,136 above.
- Whether the same fused-vs-per-branch confusion reached any other preset. The counts above are the only ones I measured against a reference.

Measured against `main` @ eeab195, with `src/olm/models` and `src/olm/nn` verified byte-identical to `origin/main`, on torch 2.2.2 / Python 3.12 / CPU.
